packages/integration-sdk-http-client/src/index.ts

```diff
@@ -0,0 +1,86 @@
+import fetch from 'node-fetch';
+import { getUnixTimeNow, sleep, isNonRetryableError } from './util';
+import {
+  RateLimitConfig,
+  APIResponse,
+  RateLimitState,
+  APIRequest,
+} from './types';
+
+export const DEFAULT_RATE_LIMIT_CONFIG: RateLimitConfig = {
+  maxAttempts: 5,
+  reserveLimit: 30,
+  cooldownPeriod: 1000,
+  sleepAdditionalSeconds: 0,
+};
+
+export type APIResourceIterationCallback<T> = (
+  resources: T[],
+) => boolean | void | Promise<boolean | void>;
+
+export class APIClient {
+  private rateLimitConfig: RateLimitConfig = DEFAULT_RATE_LIMIT_CONFIG;
+  private rateLimitState: RateLimitState;
+
+  async executeAPIRequest(request: APIRequest): Promise<APIResponse> {
+    let attempts = 0;
+
+    do {
+      const response = await fetch(request.url, request);
+
+      if (isNonRetryableError(response.status)) {
+        const err = new Error(
+          `API request error for ${request.url}: ${response.statusText}`,
```

Comment:
Might be beneficial to do 
`API request error for ${request.url}: ${response.code} ${response.statusText}`,